File: services/comments/internal/domain/comments.go
package domain

type Comment struct {
	Id      string `json:"id"`
	Content string `json:"content"`
	Status  string `json:"status"`
}

type CreateCommentReq struct {
	Content string `json:"content"`
}

type GetCommentsResp struct {
	Comments []*Comment `json:"comments"`
}

type CommentWithPostId struct {
	PostId string `json:"postId"`
	Comment
}

File: web/src/api/comments.ts
import axios from "axios";
import type { comment, comments } from "./types";

const comments_url = "http://localhost:4001";

export async function createComment(
	postId: string,
	content: string,
): Promise<comment> {
	const req = await axios.post(`${comments_url}/posts/${postId}/comment`, {
		content,
	});
	return req.data;
}

export async function getComments(postId: string): Promise<comments> {
	const req = await axios.get(`${comments_url}/posts/${postId}/comments`);
	return req.data;
}
